tests/processing/test_processor.py

```diff
@@ -150,7 +150,9 @@ def test_stream_processor_lifecycle() -> None:
         (Timing, "arroyo.consumer.shutdown.time"),
         (Timing, "arroyo.consumer.callback.time"),
         (Timing, "arroyo.consumer.poll.time"),
+        (Increment, "arroyo.consumer.pause"),
```

Comment:
random thought: i think this test is kind of bs and annoying, but at teh same time it's somewhat interesting to see that we apply backpressure when a consumer is shutting down